truffle/src/main/java/org/jruby/truffle/nodes/core/ArrayNodes.java

```diff
@@ -446,10 +449,11 @@ public Object setIntegerFixnumRange(VirtualFrame frame, RubyArray array, RubyRan
 
     }
 
-    @CoreMethod(names = "at", required = 1, lowerFixnumParameters = 0)
```

Comment:
Per my comment on the PR, I think this should stay and a new specialization for `double` added.